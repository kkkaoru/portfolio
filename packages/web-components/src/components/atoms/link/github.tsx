import { ReactComponent as GithubLogo } from '@web-components/assets/brand/github/black.svg';
import { LinkIncludedChildren } from './link-included-children';
import { generateGitHubUrl } from './functions';
import type { SnsLinkProps } from './types';

export function GitHubLink({ id, ariaLabel = 'GitHub', ...linkProps }: SnsLinkProps): JSX.Element {
  const gitHubUrl = generateGitHubUrl(id);
  if (gitHubUrl === undefined) {
    return <></>;
  }
  return (
    <LinkIncludedChildren href={gitHubUrl} ariaLabel={ariaLabel} {...linkProps}>
      <GithubLogo />
    </LinkIncludedChildren>
  );
}
