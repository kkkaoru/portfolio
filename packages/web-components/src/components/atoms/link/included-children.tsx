export type LinkIncludedChildrenProps = {
  href: string;
  ariaLabel: string;
  children: React.ReactNode;
  isReversedAnimation?: boolean;
  target?: string;
  rel?: string;
};

export function LinkIncludedChildren({
  href,
  ariaLabel,
  children,
  isReversedAnimation = false,
  target = '_blank',
  rel = 'noopener',
}: LinkIncludedChildrenProps): JSX.Element {
  const animation = isReversedAnimation ? 'animate-down-up-down' : 'animate-up-down-up';
  return (
    <a href={href} target={target} rel={rel} className={`${animation} block w-16`} aria-label={ariaLabel}>
      {children}
    </a>
  );
}
