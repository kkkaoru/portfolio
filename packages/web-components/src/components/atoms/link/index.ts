export * from './link-included-image';
