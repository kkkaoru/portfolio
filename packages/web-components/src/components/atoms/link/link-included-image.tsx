export type LinkAndImageProps = {
  href: string;
  src: string;
  alt: string;
  isReversedAnimation?: boolean;
  width?: number;
  height?: number;
  target?: string;
  rel?: string;
  loading?: 'eager' | 'lazy';
  decoding?: 'async' | 'auto' | 'sync';
};

export function LinkIncludedImage({
  href,
  src,
  alt,
  width = 128,
  height = 128,
  isReversedAnimation = false,
  target = '_blank',
  rel = 'noopener',
  loading = 'lazy',
  decoding = 'async',
}: LinkAndImageProps): JSX.Element {
  const animation = isReversedAnimation ? 'animate-down-up-down' : 'animate-up-down-up';
  return (
    <a href={href} target={target} rel={rel}>
      <img
        src={src}
        alt={alt}
        width={width}
        height={height}
        loading={loading}
        decoding={decoding}
        className={`${animation} hover:animation-paused`}
      />
    </a>
  );
}
