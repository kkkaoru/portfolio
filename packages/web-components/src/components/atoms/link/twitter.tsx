import { ReactComponent as TwitterLogo } from '@web-components/assets/brand/twitter/blue.svg';
import { LinkIncludedChildren } from './link-included-children';
import { generateTwitterUrl } from './functions';
import type { SnsLinkProps } from './types';

export function TwitterLink({ id, ariaLabel = 'Twitter', ...linkProps }: SnsLinkProps): JSX.Element {
  const twitterUrl = generateTwitterUrl(id);
  if (twitterUrl === undefined) {
    return <></>;
  }
  return (
    <LinkIncludedChildren href={twitterUrl} ariaLabel={ariaLabel} {...linkProps}>
      <TwitterLogo />
    </LinkIncludedChildren>
  );
}
