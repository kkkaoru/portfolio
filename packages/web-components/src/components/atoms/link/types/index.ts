export * from './sns';
export * from './text';
