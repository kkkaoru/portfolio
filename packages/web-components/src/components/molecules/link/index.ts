export * from './list';
export * from './contact';
