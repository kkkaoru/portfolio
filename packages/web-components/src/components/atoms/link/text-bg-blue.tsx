import { LinkTextProps } from './types';

export function LinkTextBgBlue({
  text,
  target = '_blank',
  rel = 'noopener',
  ...linkProps
}: LinkTextProps): JSX.Element {
  return (
    <a
      {...linkProps}
      target={target}
      rel={rel}
      className="inline-block font-sans text-base font-bold leading-none border-0 cursor-pointer rounded-[3em] py-[11px] px-[20px] text-[#fff] bg-[#1ea7fd]"
    >
      {text}
    </a>
  );
}
