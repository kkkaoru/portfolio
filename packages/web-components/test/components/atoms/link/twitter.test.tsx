import { render } from '@testing-library/react';
import { TwitterLink } from '@web-components/components/atoms/link';

describe('twitter link', () => {
  it('should be matched snapshot', () => {
    const { asFragment } = render(<TwitterLink id="TwitterJp" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
