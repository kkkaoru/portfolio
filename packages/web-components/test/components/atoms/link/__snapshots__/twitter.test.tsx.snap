// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`twitter link should be matched snapshot 1`] = `
<DocumentFragment>
  <a
    aria-label="Twitter"
    class="animate-up-down-up flex w-16"
    href="https://twitter.com/TwitterJp"
    rel="noopener"
    target="_blank"
  >
    <svg
      data-jest-file-name="blue.svg"
      data-jest-svg-name="blue"
      data-testid="blue"
    />
  </a>
</DocumentFragment>
`;
